ARCHITECTURE.md

```diff
@@ -0,0 +1,194 @@
+
+At the time of writing, almost a year of exploratory and freestyle project history has passed.
+Fortunately, it is not necessary for further understanding, so I'll spare you that.
+This document, though, lays out the design, architecture and direction of Standard.
+
+If the topics discussed herein are dear to you, please take it as an invitation to get involved.
+
+This design document shall be stable and amendments go through a proper process of consideration.
+
+## Overview
+
+Standard is a collection of functionality and best practices (_"framework"_) to bootstrap and sustain the automatable sections of the Software Delivery Lifecycle (SDLC) _efficiently_ with the power of Nix and Flakes.
+In particular, Standard is a _Horizontal\* Integration Framework_ which integrates _vertical\*_ tooling.
+
+> <sub>We occasionally adapt concepts from non-technical contexts. This is one instance.</sub>
+>
+> _Vertical Tooling_ does one thing and does it well in a narrow scope (i.e "vertical").
+>
+> _Horizontal Tooling_ stitches vertical tooling together to a polished whole.
+
+What is being integrated are the end-to-end automatable sections of the SDLC.
+For these we curate a collection of functionality, tools and best practices.
+
+An SDLCs _efficiency_ on the is characterized by two things.
+
+Firstly, by adequate _lead time_ which is the amount of time it takes to set up an initial version of the software delivery pipeline.
+It needs to be _adequate_ rather than _just fast_, as it takes place in the context of a team.
+Rather than for speed, they need optimization for success.
+For example, a process needs to be documented & explained and your team needs to be trained on it.
+Standard encourages incremental adoption in order to leave enough space for these paramount activities.
+If you're in a hurry and your team is onboard, though, you still can jumpstart the implementation.
+
+Secondly, efficient SDLCs are characterized by short _cycle times_ which is the amount of time it takes for a designed feature to be shipped to production.
+Along this journey, we encounter our scope (more on it below):
+
+- aspects of the _development_ environment;
+- the packaging pipeline that produces artifacts;
+- and continuous processes integrating the application lifecycle.
+
+Hence, the goal of Standard is to:
+
+- Enable easy and incremental adoption
+- Optimize the critical path that reduces your SDLC's cycle time.
+
+Additionally, unlike similar projects, we harness the power of Nix & Flakes to ensure reproducibility.
+
+## Goals
+
+- _Complete_: Standard should cover the important use cases for setting up and running the automatable sections of the SDLC.
+- _Optimized_: Standard should optimize both for the needs of the individual developers and the market success of the product.
+- _Integrated_: Standard should provide the user with a satisfying integration experience across a well-curated assortment of tools and functionality.
```

Comment:
maybe add:
with the possibility of extending this functionality seemlessly when necessary